spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -276,13 +286,21 @@ else if (messageListener instanceof MessageListener) {
 			throw new IllegalStateException("messageListener must be 'MessageListener' "
 					+ "or 'AcknowledgingMessageListener', not " + messageListener.getClass().getName());
 		}
-		if (getTaskExecutor() == null) {
-			setTaskExecutor(
-					new SimpleAsyncTaskExecutor(getBeanName() == null ? "kafka-" : (getBeanName() + "-kafka-")));
+		if (getConsumerTaskExecutor() == null) {
+			SimpleAsyncTaskExecutor pooledExecutor = new SimpleAsyncTaskExecutor(
```

Comment:
It shouldn't be a big deal since we're not starting/stopping consumers regularly - the listener executor is indeed pooled for this very reason.

I can change it if you insist.